### ISSUE TITLE
Semantic of `this` with dynamic variables and layered environments

### ISSUE THREAD

**Issue by ixemad** [dynaconf/dynaconf#1030]
Basically, In [Dynamic variables](https://www.dynaconf.com/dynamic/) documentation says that _`{this.current_env}` is the same as `settings.current_env`_ but it is not when using it with multi-layered environments. For instance in the following scenario

```
[ default]
region = ''
product =  '@format arn://{this.region}'

[development]
region = 'us-west-2'
```
When I load the configuration with the `development` profile, `settings.region` is `us-west-2` but `setttings.product` is `arn://`. It should have to be `arn://us-west-2` in my understanding.

**Comment by rochacbruno** [comment]
IS this using the latest version?

There was a bug about order of evaluation on lazy variables fixed on #933

**Comment by ixemad** [comment]
I thought so but it wasn't. A failed poetry update I didn't notice :pray:. Now it works as expected. 

Blazing fast response. Simply amazing :clap: :clap:  :clap:
